aldryn_django/middleware.py

```diff
@@ -1,4 +1,48 @@
 from django.conf import settings
+from django.core.urlresolvers import is_valid_path, get_script_prefix
+from django.http import HttpResponsePermanentRedirect
+from django.utils.translation import get_language_from_path
+
+
+class LanguagePrefixFallbackMiddleware(object):
+    """
+    Redirects urls which raise a 404 and contain a language prefix
+    that matches the default language to their non-language prefix
+    version.
+    """
+
+    response_redirect_class = HttpResponsePermanentRedirect
```

Comment:
``PermanentRedirects`` make me nervous. If someone enables the redirect for a few days and then turns it off again the browsers of the users that visited the website will have cached the redirect... and end up in a redirect loop.

I guess the only way around this would be to make it configurable and encourage people to use the ``TemporaryRedirect`` for a while, until they are really sure this is the right thing.